ecosystem/sep-0010.md

```diff
@@ -18,57 +18,58 @@ This SEP defines the standard way for clients such as wallets or exchanges to cr
+1. The **Client Application** verifies that the transaction has an invalid sequence number 0.  This is extremely important to ensure the transaction isn't malicious.
+1. The **Client Application** verifies that the transaction is signed by the `SIGNING_KEY` obtained through discovery flow.
+1. The **Client Application** verifies that the transaction's first operation is a Manage Data operation that has its:
+   1. Source account set to the **Authenticating Account**
+   1. Key set to `<home domain> auth` where the home domain is the **Home Domain**.
    1. Value set to a nonce value.
-1. The client verifies that if the transaction has a Manage Data operation with key `web_auth_domain` that it has:
-   1. Source account set to the server's account.
-   1. Key set to `web_auth_domain`.
-   1. Value set to the domain name of the SEP-10 server from which the client requested the challenge.
-1. The client verifies that if the transaction has other operations they are Manage Data operations that all have their source accounts set to the the server's account.
-1. The client signs the transaction using the secret key(s) of signers for the user's Stellar account
-1. The client submits the signed challenge back to the server using [`token`](#token) endpoint
-1. The server checks that the user's account exists
-1. If the user's account exists:
-  1. The server gets the signers of the user's account
-  1. The server verifies the client signatures count is one or more;
-  1. The server verifies the client signatures on the transaction are signers of the user's account;
-  1. The server verifies the weight provided by the signers meets the required threshold(s), if any
-  1. If the signatures check out, the server responds with a [JWT](https://jwt.io) that represents the user's session
-1. If the user's account does not exist (optional):
-  1. The server verifies the client signature count is one
-  1. The server verifies the client signature is correct for the master key of the account
-1. Any future calls to the server can be authenticated by including the JWT as a parameter
+1. The **Client Application** verifies that if the transaction has a Manage Data operation with key `web_auth_domain` that it has:
+   1. Source account set to the ``SIGNING_KEY`` from the **Home Domain**'s [SEP-1 stellar.toml](sep-0001.md).
+   1. Value set to the **Web Auth Domain**.
+1. The **Client Application** verifies that if the transaction has other operations they are Manage Data operations that all have their source accounts set to the the ``SIGNING_KEY`` from the **Home Domain**'s [SEP-1 stellar.toml](sep-0001.md).
+1. The **Client Application** signs the transaction using the secret key(s) of the signer(s) for the **Authenticating Account**
+1. The **Client Application** submits the signed challenge back to the **Web Auth Endpoint** using [`token`](#token) endpoint
+1. The **Web Auth Endpoint** checks that the **Authenticating Account** exists
```

Comment:
I think having a `Web Auth Endpoint`, `Web Auth Server`, and `Web Auth Domain` adds more potential for confusion than having a single `Web Auth` participant and referencing related properties of that participant. Listing each as participants makes it seem like they are independent of each other, even if the definitions at the top of the document clarify that they're all related.

My preference would be to only define `Web Auth Server` in the participants list, but as you suggest we can make sure we're referring to the `Web Auth Server`'s endpoint when appropriate, and same for it's domain.

Reply:
I see your point. 👍

I think this also raises the question if the authenticating account is necessary, given that it is just a component of the client and it might be better to return to Client and Server, but do as you've done here and make them explicit identifiers by holding them and using them intentionally. Then attribution introduces only one new concept, the Client Server.

Thoughts?

Reply:
So you're suggesting to have the following components defined in the abstract:
- Home Domain
- Client Application
- Web Auth Server

But still use the bolded style when referring to properties related to the components defined in the abstract? I think we should reserve the bolded style to explicitly defined components, because bolding effectively implies that they are defined.

Do you think any of the following phrases are ambiguous?

> Source account set to the account the **Client Application** is authenticating

> The **Client Application** obtains a unique [`challenge`](#challenge) from the endpoint implemented by the **Web Auth Server**

> The source account is the **Home Domain**'s ``SIGNING_KEY``, the value is the domain on which the **Web Auth Server** is hosted.

The above phrases replaces usage of **Authenticating Account**, **Web Auth Endpoint**, and **Web Auth Domain** terms. It may be more verbose but I don't think there is potential for misunderstanding.

Client Attribution would then introduce the **Client Server**, along with language like the following:

> The **Client Server** should implement an endpoint that accepts a challenge transaction from the **Client Application** and signs it using the `SIGNING_KEY` listed on the [SEP-1 stellar.toml](...) file hosted on it's domain

Reply:
I was meaning what you outlined, only three bolded components and rely on as brief identification for others, then the client server becomes a new component.

I'd like to find a more succinct form for the server account/signing key though. 

I think if we rename Web Auth Server to simply Server, then to say Client account and Server account is sufficient, then Client Server is still distinct?

Reply:
Sounds good to me!